atoma-state/src/handlers.rs

```diff
@@ -1571,15 +1571,49 @@ pub async fn handle_node_key_rotation_event(
         evidence_bytes,
     } = event;
     let original_evidence_bytes = decompress_bytes(&evidence_bytes)?;
-    let evidence_data = serde_json::from_slice::<Vec<DeviceEvidence>>(&original_evidence_bytes)?;
-    let is_valid = remote_attestation_verification::attest_nvidia_evidence_list(
-        state_manager,
-        &evidence_data,
-        &new_public_key,
-        device_type,
-    )
-    .await
-    .is_ok();
+    let evidence_data = serde_json::from_slice::<Vec<CombinedEvidence>>(&original_evidence_bytes)?;
+    let mut gpu_evidence_data = Vec::new();
+    let mut nvswitch_evidence_data = Vec::new();
+    for evidence in evidence_data {
+        match evidence {
+            CombinedEvidence::Device(evidence) => gpu_evidence_data.push(evidence),
+            CombinedEvidence::NvSwitch(evidence) => nvswitch_evidence_data.push(evidence),
+        }
+    }
+    let is_ppcie_mode = gpu_evidence_data.len() == 8 && nvswitch_evidence_data.len() == 4;
+    let is_valid = if is_ppcie_mode {
+        let mut is_valid = remote_attestation_verification::attest_nvidia_gpu_evidence_list(
+            state_manager,
+            &gpu_evidence_data,
+            &new_public_key,
+            device_type,
+        )
+        .await
+        .is_ok();
```

Comment:
using `is_ok()` discards specific error information that could be valuable for debugging or logging.

Reply:
True, will refactor